kiwi/builder/disk.py

```diff
@@ -476,7 +486,8 @@ def __write_bootloader_config_to_system_image(self, device_map):
         boot_uuid = self.disk.get_uuid(
             boot_device.get_device()
         )
-        self.bootloader_config.setup_disk_boot_images(boot_uuid)
+        if self.firmware.efi_mode():
+            self.bootloader_config.setup_disk_boot_images(boot_uuid)
```

Comment:
Could you remove this condition ? the bootloader_config object should know if it needs to create images or not.
The more conditions the harder to test and the harder to debug. Thanks